tests/baselines/reference/recursiveResolveTypeMembers.errors.txt

```diff
@@ -1,16 +1,13 @@
-tests/cases/compiler/recursiveResolveTypeMembers.ts(4,49): error TS2577: Return type annotation circularly references itself.
 tests/cases/compiler/recursiveResolveTypeMembers.ts(4,58): error TS2304: Cannot find name 'H'.
 tests/cases/compiler/recursiveResolveTypeMembers.ts(4,62): error TS2574: A rest element type must be an array type.
 tests/cases/compiler/recursiveResolveTypeMembers.ts(4,79): error TS2304: Cannot find name 'R'.
 
 
-==== tests/cases/compiler/recursiveResolveTypeMembers.ts (4 errors) ====
+==== tests/cases/compiler/recursiveResolveTypeMembers.ts (3 errors) ====
     // Repro from #25291
     
     type PromisedTuple<L extends any[], U = (...args: L) => void> =
         U extends (h: infer H, ...args: infer R) => [Promise<H>, ...PromisedTuple<R>] ? [] : []
-                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Comment:
I assume that all these circular reference errors go away because the types are now deferred? Does this example end up with no error because the rest of the program never uses PromisedTuple, or because the error was not necessary in the first place?

Reply:
It goes away because the type is deferred now, yes. `getReturnTypeOfSignature` needs to be recurred into to trigger the error - but now, we ask for the return type, get the conditional, attempt to simplify the conditional, see the conditional instantiate itself into itself as part of one of its branch types, then go "nope, definitely won't simplify" and give it its deferred type, which is then what we return. Since we never actually inspect the resolved type of that inner instantiation (since we went "ah, it's going to be the same as one that's already resolving"), we never assigned it the circularity error.